### PR TITLE
[Feature] Add firebase to ios project

### LINKED ISSUE #18: Add firebase to iOS project

### PR DESCRIPTION
**Pull request by mar-cas3**
Followed this guide: https://firebase.google.com/docs/ios/setup

* Adds google configuration file + SDK packages as dependencies to project.
* Configures the entry point of the app to use Firebase (see [FoodBookApp.swift](./FoodBookApp/FoodBookApp.swift))

Additionally, configured to use Analytics packages for future implementation. 
Screenshot 2024-02-24 at 9 18 53 PM

Closes #18

### CODE REVIEW

**Comment by mar-cas3** [comment]
Looking into merge conflict...

File: FoodBookApp/FoodBookApp.swift [hunk 1]
```diff
@@ -6,9 +6,22 @@
...
+    FirebaseApp.configure()
+
+    return true
+  }
+}
 
 @main
 struct FoodBookApp: App {
+    //  register app delegate for Firebase setup
+    @UIApplicationDelegateAdaptor(AppDelegate.self) var delegate
```
Comment (juanyepesp) [inline FoodBookApp/FoodBookApp.swift:23]: How can we use this? Maybe document for future reference?

**Review by juanyepesp — CHANGES_REQUESTED** [review]
Pls include usage of the SDK. Thanks!

**Review by juanyepesp — APPROVED** [review]
awesome! thanks for including example. 😄

**Pull request merged**